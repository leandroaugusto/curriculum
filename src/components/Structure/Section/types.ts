export interface SectionProps {
  readonly className?: string;
  readonly title?: string;
  readonly children: React.ReactNode;
}
