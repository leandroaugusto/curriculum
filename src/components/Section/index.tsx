interface SectionProps {
  readonly className?: string;
  readonly children: React.ReactNode;
  readonly title: string;
}

export default function Section({ children, title, className }: SectionProps) {
  return (
    <section className={`text-gray-800 ${className ?? ""}`}>
      <h3 className="font-bold uppercase bg-gray-200 px-2">{title}</h3>
      <div className="px-2">{children}</div>
    </section>
  );
}
