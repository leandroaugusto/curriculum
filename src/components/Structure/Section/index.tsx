import { SectionProps } from "./types";

export default function Section({ children, title, className }: SectionProps) {
  return (
    <section className={`text-gray-800 ${className || ""}`}>
      {title && (
        <h3 className="font-bold uppercase bg-gray-200 px-4">{title}</h3>
      )}
      <div className="px-4">{children}</div>
    </section>
  );
}
