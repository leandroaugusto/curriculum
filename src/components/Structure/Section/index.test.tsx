import { render } from "@testing-library/react";

import Section from ".";

describe("Section", () => {
  it("renders the Section component", () => {
    const { container } = render(
      <Section>
        <p>Test</p>
      </Section>
    );

    expect(container).toMatchSnapshot();
  });
});
