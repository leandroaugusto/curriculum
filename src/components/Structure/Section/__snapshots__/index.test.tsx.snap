// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Section renders the Section component 1`] = `
<div>
  <section
    class="text-gray-800 "
  >
    <div
      class="px-4"
    >
      <p>
        Test
      </p>
    </div>
  </section>
</div>
`;
